samza-core/src/main/scala/org/apache/samza/job/JobRunner.scala

```diff
@@ -82,12 +82,12 @@ class JobRunner(config: Config) extends Logging {
     val jobFactory: StreamJobFactory = getJobFactory
     val coordinatorSystemConsumer = new CoordinatorStreamSystemConsumer(config, new MetricsRegistryMap)
     val coordinatorSystemProducer = new CoordinatorStreamSystemProducer(config, new MetricsRegistryMap)
+    val systemAdmins = new SystemAdmins(config)
 
     // Create the coordinator stream if it doesn't exist
     info("Creating coordinator stream")
     val coordinatorSystemStream = CoordinatorStreamUtil.getCoordinatorSystemStream(config)
-    val systemFactory = CoordinatorStreamUtil.getCoordinatorSystemFactory(config)
-    val systemAdmin = systemFactory.getAdmin(coordinatorSystemStream.getSystem, config)
+    val systemAdmin = systemAdmins.getSystemAdmin(coordinatorSystemStream.getSystem)
```

Comment:
Could you rename this var to coordinatorSystemAdmin? Might help prevent someone else from using the wrong systemAdmin in the future.

Reply:
Renamed variables and added a ConfigException throw.